File: bench/promptgreen/tasks/m04/hidden_tests.mjs
// Hidden test for m04 (boundary check). Never shown to an author.
const EXPECTED = '1\n0\n';

export async function run(compileFn, source) {
  const r = compileFn(source);
  const lines = (r.stdout || '').split('\n');
  const checks = [];
  checks.push(r.ok === true);
  checks.push(r.stdout === EXPECTED);
  checks.push(lines[0] === '1');
  const green = checks.every(Boolean);
  return { green, detail: { expected: EXPECTED, got: r.stdout, ok: r.ok, checks } };
}

File: bench/promptgreen/tasks/m04/py/hidden_tests.py
import os
import subprocess
import sys

HERE = os.path.dirname(os.path.abspath(__file__))
EXPECTED = "1\n0\n"


def main() -> int:
    result = subprocess.run(
        [sys.executable, os.path.join(HERE, "reference.py")],
        capture_output=True,
        text=True,
    )
    assert result.returncode == 0, f"nonzero exit: {result.returncode}"
    assert result.stdout == EXPECTED, f"got {result.stdout!r}, expected {EXPECTED!r}"
    assert result.stdout.split("\n")[0] == "1"
    print("m04 ok")
    return 0


if __name__ == "__main__":
    sys.exit(main())
